src/shape_inference.jl

```diff
@@ -352,6 +352,13 @@ for func in ["Sum", "Prod", "Min", "Max", "All", "Any", "Mean"]
     end
 end
 
+register_shape("UnsortedSegmentSum") do op
+    # TODO handle case of partial reduction
+    value_shape = copy(_get_shape(get_input(op, 1)))
+    value_shape.dims[1] = get_attr(op, "num_segments", Int64)
```

Comment:
I *think* (not 100% sure).
That this line should be:

    value_shape.dims[1] = load_const(get_input(op, 2))

as `"num_segments"` is an Input, not an attribute.